src/Application.php

```diff
@@ -25,14 +25,14 @@ class Application implements MiddlewareInterface, RequestHandlerInterface
     /** @var RouteCollectorInterface */
     private $routes;
 
-    /** @var RequestHandlerRunner */
+    /** @var RequestHandlerRunnerInterface */
     private $runner;
 
     public function __construct(
         MiddlewareFactory $factory,
         MiddlewarePipeInterface $pipeline,
         RouteCollectorInterface $routes,
-        RequestHandlerRunner $runner
+        RequestHandlerRunnerInterface $runner
```

Comment:
Looks like a major BC break though :scream:

Reply:
I optimistically assumed that the interface was compatible with the implementation :innocent: 

... is that not the case ?

Reply:
Because `RequestHandlerRunner` was made final, it could not be mocked in tests anymore. So I changed the mocks to mock the interface instead. But then of course the application was throwing error because it as still expecting an instance of the final `RequestHandlerRunner`...

@Ocramius, would you have a recommendation on how to push this forward ? It feels a bit overkill to have a major release only for that. But I don't really see a clean way out of it, unless if we remove the `final` from `RequestHandlerRunner` ?

Reply:
Turns out we can use a real `RequestHandlerRunner` with mocked parameters and pass all tests without other modifications. So we don't break anything anymore :heart_eyes:

Reply:
AAAAAAAHHHHH this is a total brain-fart by me, sorry @PowerKiKi 

Looking at a type-change in a method parameter, I immediately thought of a BC break, but `__construct()` is not affected by such semantics: I'm afraid I made you do much more legwork than necessary here, and the type change was perfectly valid, as it is contravariant, sorry :(

Reply:
No worries, here is the simpler version force pushed again